File: e-commerce-frontend/src/types/domains/variation.ts
export interface Variation {
    variationId: number;
    name: string;
    variationOptions: VariationOption[];
}

export interface VariationOption {
    variationOptionId: number;
    name: string;
    code: string;
}

export interface VariationCreationPayload {
    name: string;
    variationOptions: {
        name: string;
        code: string;
    }[];
}

export interface VariationUpdationPayload {
    name: string;
    variationOptions: {
        variationOptionId?: number;
        name: string;
        code: string;
    }[];
}

File: e-commerce-backend/src/main/java/com/TechKun/model/Variation.java
package com.TechKun.model;

import java.util.List;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@Table(name = "Variation")
public class Variation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "variation_id", nullable = false)
    private Integer variationId;

    @Column(name = "name", nullable = false)
    private String name;

    @OneToMany(mappedBy = "variation", orphanRemoval = true, cascade = CascadeType.ALL)
    private List<VariationOption> variationOptions;

    public Variation(Integer variationId) {
        this.variationId = variationId;
    }
}
